File: frontend/lib/mail/templates/system-expired.ts
import { renderEmailLayout, es } from './layout'

export const SystemExpiredTemplate = (
  adminName: string,
  systemName: string,
  hostName: string,
  ipAddress: string,
  orgName: string,
  domain: string
) => {
  const content = `
    <h2 style="${es.h1}">Terminal token expired</h2>
    <p style="${es.p}">
      Hi ${adminName}, the security token for a voting terminal in <strong style="${es.strong}">${orgName}</strong> has expired and the terminal has been automatically disconnected.
    </p>

    <div style="${es.card('#ef4444')}">
      <p style="margin:0 0 12px 0;font-size:13px;font-weight:700;text-transform:uppercase;letter-spacing:0.07em;color:#94a3b8;">Terminal Details</p>
      <p style="${es.row}"><strong style="${es.strong}">Name:</strong> ${systemName}</p>
      <p style="${es.row}"><strong style="${es.strong}">Hostname:</strong> ${hostName || 'N/A'}</p>
      <p style="${es.row}"><strong style="${es.strong}">IP Address:</strong> ${ipAddress || 'N/A'}</p>
      <p style="margin:0;font-size:14px;color:#475569;"><strong style="${es.strong}">Organization:</strong> ${orgName}</p>
    </div>

    <div style="${es.warningBox}">
      <p style="margin:0;font-size:14px;color:#92400e;">
        &#9888;&nbsp; This terminal can no longer authorize votes or sync data until it is re-approved by an administrator.
      </p>
    </div>

    <p style="${es.p}">
      To restore service, go to the dashboard, select this terminal, and click <strong style="${es.strong}">Approve</strong> to generate a fresh security token.
    </p>

    <div style="${es.cta}">
      <a href="${domain}/admin/organization/systems" style="${es.button('#ef4444')}">Re-authorize Terminal</a>
    </div>
  `
  return renderEmailLayout(content, `Terminal Token Expired: ${systemName} — E-Voting`, 'warning')
}

File: backend/app/services/mail/templates/system_expired.py
from .layout import render_email_layout

def render_system_expired_email(
    admin_name: str,
    system_name: str,
    host_name: str,
    ip_address: str,
    org_name: str,
    systems_url: str
) -> str:
    content = f"""
    <div class="greeting">Hello {admin_name or "Administrator"},</div>
    <div class="message">
        The security token for the terminal <strong>{system_name}</strong> in <strong>{org_name}</strong> has expired.
        The terminal is now offline and requires re-authorization to resume operations.
    </div>
    
    <div class="details-container">
        <div class="detail-row">
            <span class="detail-label">Terminal Name:</span>
            <span>{system_name}</span>
        </div>
        <div class="detail-row">
            <span class="detail-label">Host Name:</span>
            <span>{host_name}</span>
        </div>
        <div class="detail-row">
            <span class="detail-label">IP Address:</span>
            <span>{ip_address}</span>
        </div>
        <div class="detail-row">
            <span class="detail-label">Status:</span>
            <span style="color: #ef4444; font-weight: 700;">EXPIRED</span>
        </div>
    </div>
    
    <div class="button-container">
        <a href="{systems_url}" class="button">Re-authorize Terminal</a>
    </div>
    
    <div class="message">
        If this terminal is no longer in use, you can safely ignore this notification or revoke the access permanently from the dashboard.
    </div>
    """
    return render_email_layout(content, title="E-Voting Security")
